packages/mui-material/src/SwipeableDrawer/SwipeableDrawer.d.ts

```diff
@@ -2,6 +2,20 @@ import * as React from 'react';
 import { DrawerProps } from '../Drawer';
 
 export interface SwipeableDrawerProps extends Omit<DrawerProps, 'onClose' | 'open'> {
+  /**
+   * Callback to determine what children of the drawer the user can use to drag open the drawer.
+   * Can be a custom function to control the behavior.
+   * Set or return true / false to allow / disallow the swipe event.
```

Comment:
```suggestion
   * If set to true the swipe event will open the drawer even if it starts over some of its children.
   * It can be useful in scenarios where the drawer is partially visible.
   * Can be further customized by a callback to determine which children the user can drag over to open the drawer (for e.g. to ignore other elements that handle the touch move events, like sliders).
```

Reply:
I've restructured a bit to start with the "most common" usage first. Then I explained a possible scenario, and then finished with the least common use-case. cc @samuelsycamore for a final checkup on the wording.

Reply:
cc @samuelsycamore pinging you again

Reply:
Sorry I missed this the first time! I'm not sure exactly what is meant here:

> even if it starts over some of its children

Is there another way to phrase this?

Reply:
It is related to the drag event. You started dragging over some children element. Considering your comment, this would be confusing for other users of the library too :\

Reply:
@michaldudak maybe you can help here to unblock the PR, the only thing left as far as I remember is the prop descriotion. Sam is on PTO, so I don't expect him to respond, and the PR has been sitting for quite some time.

Reply:
I'm back! 😁 How does this sound?

```suggestion
   * If set to true, the swipe event will open the drawer even if the user begins the swipe on one of the drawer's children.
   * This can be useful in scenarios where the drawer is partially visible.
   * You can customize it further with a callback that determines which children the user can drag over to open the drawer (for example, to ignore other elements that handle touch move events, like sliders).
```

Reply:
Sam's suggestion looks good. I merged in the latest master and fixed all the conflicts. It should be good to go now.